src/main/java/org/jenkinsci/plugins/workflow/cps/DSL.java

```diff
@@ -481,14 +546,18 @@ private NamedArgsAndClosure(Map<?,?> namedArgs, Closure body) {
      * but better to do it here in the Groovy-specific code so we do not need to rely on that.
      * @return {@code v} or an equivalent with all {@link GString}s flattened, including in nested {@link List}s or {@link Map}s
      */
-    private static Object flattenGString(Object v) {
+    private static Object flattenGString(Object v, @CheckForNull Set<String> interpolatedStrings) {
```

Comment:
I would make the `interpolatedStrings` parameter `@NonNull` instead and use empty sets in cases where null was being used previously.

Reply:
done in 70c4c223dd4d08ca065fb9f8b0e41bb3ccc07c6d